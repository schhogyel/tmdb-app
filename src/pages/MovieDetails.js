import React from "react";

import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import { ArrowBack } from "@material-ui/icons";
import { Link } from "@reach/router";

import MovieDetailsCard from "../components/MovieDetailsCard";
import Typography from "@material-ui/core/Typography";

import { fetchMovie } from "../api";

const styles = theme => ({
  heroImage: {
    maxHeight: "300px",
    overflow: "hidden",
    display: "flex"
  },
  image: {
    width: "100%",
    objectFit: "cover"
  },
  backArrow: {
    position: "absolute",
    left: 0,
    width: "2.5rem",
    height: "2.5rem",
    color: "#fff"
  },
  detailsContainer: {
    maxWidth: "1200px",
    margin: "auto",
    padding: "10px"
  },
  overviewDescription: {
    color: "#B8D8E6",
    textAlign: "left"
  }
});

function MovieDetails(props) {
  const { id, classes } = props;
  const [movie, setMovie] = React.useState({});
  React.useEffect(() => {
    let abort = false;
    fetchMovie(id, abort).then(response => setMovie(response));
    return () => {
      abort = true;
    };
  }, [id]);
  return (
    <main>
      <Link to={"/"}>
        <ArrowBack className={classes.backArrow} />
      </Link>

      <div className={classes.heroImage}>
        <img
          className={classes.image}
          src={`https://image.tmdb.org/t/p/w500/${movie.backdrop_path}`}
          alt="poster"
        />
      </div>
      <div className={classes.detailsContainer}>
        <MovieDetailsCard
          title={movie.title}
          year={movie.release_date}
          score={movie.vote_average}
          image={movie.poster_path}
          duration={movie.runtime}
        />
        <Typography className={classes.overviewDescription} component="p">
          {movie.overview}
        </Typography>
      </div>
    </main>
  );
}

MovieDetails.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(MovieDetails);
