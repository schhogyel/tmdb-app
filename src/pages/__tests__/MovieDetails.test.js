import React from "react";
import { shallow } from "enzyme";

import { MovieDetails } from "../MovieDetails";

describe("MovieDetails", () => {
  it("renders", () => {
    const props = { classes: {} };
    const wrapper = shallow(<MovieDetails {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
