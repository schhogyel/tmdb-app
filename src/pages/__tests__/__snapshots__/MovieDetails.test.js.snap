// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders 1`] = `ShallowWrapper {}`;
